lib/nerves_hub_link/update_manager.ex

```diff
@@ -145,26 +163,48 @@ defmodule NervesHubLink.UpdateManager do
   end
 
   # messages from Downloader
-  def handle_info({:download, :complete}, state) do
+  def handle_info({:download, :complete, _fwup_public_keys, reporting_pid}, state) do
     Logger.info("[NervesHubLink] Firmware Download complete")
-    {:noreply, %State{state | download: nil}}
+    send(reporting_pid, {:update_manager, :complete})
+    {:noreply, %State{state | status: :idle}}
+  end
+
+  def handle_info(
+        {:download, {:error, %Mint.HTTPError{reason: {:http_error, 401}}}, _fwup_public_keys,
+         reporting_pid},
+        state
+      ) do
+    Logger.error("[NervesHubLink] Firmware download error: 401")
+    send(reporting_pid, {:update_manager, {:error, :download_unauthorized}})
```

Comment:
I don't think results can be a tuple since this message eventually gets pushed up the socket and would not be JSON encodable

Reply:
I've fixed this by formatting the possible response types when sending to Hub in `socket.ex`